src/main/java/frc/robot/subsystems/conveyer/Conveyer.java

```diff
@@ -0,0 +1,143 @@
+   *       a note to the shooter, or scoring a note in the trap or amp.
+   * </ul>
+   */
+  public enum ConveyerPositions {
+    NO_NOTE,
+    HOLDING_NOTE,
+    PARTIAL_NOTE
+  }
+
+  public ConveyerPositions currentNotePosition = ConveyerPositions.NO_NOTE;
+
+  private double backMotorZeroPositionMeters;
+  private double frontMotorZeroPositionMeters;
+
+  public Conveyer() {
+    SparkMaxUtils.initWithRetry(this::setUpConveyerSparks, ConveyerCal.SPARK_INIT_RETRY_ATTEMPTS);
+    backMotorZeroPositionMeters = backMotorEncoder.getPosition();
+  }
+
+  /** Does all the initialization for the sparks, return true on success */
+  private boolean setUpConveyerSparks() {
+    int errors = 0;
+    errors += SparkMaxUtils.check(frontMotor.restoreFactoryDefaults());
+    errors += SparkMaxUtils.check(backMotor.restoreFactoryDefaults());
+
+    errors += SparkMaxUtils.check(frontMotor.setIdleMode(IdleMode.kBrake));
+    errors += SparkMaxUtils.check(backMotor.setIdleMode(IdleMode.kBrake));
+
+    frontMotor.setInverted(ConveyerConstants.FRONT_MOTOR_INVERTED);
+    backMotor.setInverted(ConveyerConstants.BACK_MOTOR_INVERTED);
+
+    errors +=
+        SparkMaxUtils.check(
+            frontMotor.setSmartCurrentLimit(ConveyerCal.CONVEYER_CURRENT_LIMIT_AMPS));
+    errors +=
+        SparkMaxUtils.check(
+            backMotor.setSmartCurrentLimit(ConveyerCal.CONVEYER_CURRENT_LIMIT_AMPS));
+
+    errors +=
+        SparkMaxUtils.check(
+            SparkMaxUtils.UnitConversions.setLinearFromGearRatio(frontMotorEncoder, ConveyerConstants.FRONT_GEAR_RATIO, ConveyerConstants.CONVEYER_MOTOR_ROLLER_DIAMETER_INS));
+    errors +=
+        SparkMaxUtils.check(
+            SparkMaxUtils.UnitConversions.setLinearFromGearRatio(backMotorEncoder, ConveyerConstants.BACK_GEAR_RATIO, ConveyerConstants.CONVEYER_MOTOR_ROLLER_DIAMETER_INS));
+
+    errors += SparkMaxUtils.check(beamBreakSensorOne.enableLimitSwitch(false));
+    errors += SparkMaxUtils.check(beamBreakSensorTwo.enableLimitSwitch(false));
+    errors += SparkMaxUtils.check(beamBreakSensorThree.enableLimitSwitch(false));
+    errors += SparkMaxUtils.check(beamBreakSensorFour.enableLimitSwitch(false));
+
+    return errors == 0;
+  }
+
+  public static Command shoot(Conveyer conveyer) {
+    return new SequentialCommandGroup(
+      new InstantCommand(() -> conveyer.frontMotor.set(ConveyerCal.PREPARE_TO_SHOOT_FRONT_SPEED), conveyer),
+      new InstantCommand(() -> conveyer.backMotor.set(ConveyerCal.PREPARE_TO_SHOOT_BACK_SPEED)),
+      new InstantCommand(() -> conveyer.currentNotePosition = ConveyerPositions.PARTIAL_NOTE),
+      new WaitCommand(ConveyerCal.NOTE_EXIT_TIME_SHOOTER_SECONDS),
+      Conveyer.stop(conveyer),
+      new InstantCommand(() -> conveyer.currentNotePosition = ConveyerPositions.NO_NOTE),
+      new WaitUntilCommand(() -> conveyer.backMotorEncoder.getVelocity() < ConveyerCal.MOTOR_VELOCITY_THRESHOLD_RPM)
```

Comment:
Now that you changed the encoder to linear, this velocity will also be linear

Reply:
Changed